src/Spice86.Core/Emulator/CPU/IInstructionExecutor.cs

```diff
@@ -3,11 +3,21 @@ namespace Spice86.Core.Emulator.CPU;
 using Spice86.Core.Emulator.VM;
 
 /// <summary>
-/// The <see cref="Cpu"/>'s API contract for the <see cref="EmulationLoop"/>
+/// API contract for an executor used by the <see cref="EmulationLoop"/>, typically a CPU
 /// </summary>
 public interface IInstructionExecutor {
     /// <summary>
     /// Executes the next instruction.
     /// </summary>
     public void ExecuteNext();
+
+    /// <summary>
+    /// Signal that we are at the entry point of the program ready to start executing our very first instruction
+    /// </summary>
+    public void SignalEntry();
```

Comment:
I wonder how this will work with DOS sub-processes, COMMAND.COM / batch files, and TSRs if it's implemented someday. Just a tought for the far future.

Reply:
We could have a specific type of call for that maybe? Not sure how about the implications.
Anyway, this change is just to materialize that CPU needs to know that execution will start and stop, this is nothing new actually.